.github/workflows/test.yml

```diff
@@ -22,26 +22,35 @@ jobs:
           source activate black
           pip install black
           black --check urlchecker
-          
-          
+
+
   type_checking:
+    needs: formatting
     runs-on: ubuntu-latest
     steps:
-      - uses: jpetrucciani/mypy-check@0.930
-      - name: mypy check
-      - with:
-          path: 'urlchecker'
-        
+      - uses: actions/checkout@v2
+
+      - name: Setup mypy environment
+        run: conda create --quiet --name type_checking mypy
+
+      - name: Check types with mypy
+        run: |
+          export PATH="/usr/share/miniconda/bin:$PATH"
+          source activate type_checking
+          pip install mypy
+          mypy --install-types
```

Comment:
Noob question - we are just using mypy to run checks correct? We don’t actually need it for the library because we can use the provided typing module! Is mypy best practice for doing these checks?

Reply:
yes exactly, it is like pytest. I am not sure if it is the best tbh but it is what popped the first when I was searching types hinting in Python